### PR TITLE
convert dropwizard-views-0.7 tests from groovy to java

### PR DESCRIPTION
**Pull request by quinoant**
related to #7195

### CODE REVIEW

File: instrumentation/dropwizard/dropwizard-views-0.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/dropwizardviews/ViewRenderTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.dropwizardviews; /*
+                                                                     * Copyright The OpenTelemetry Authors
+                                                                     * SPDX-License-Identifier: Apache-2.0
+                                                                     */
```
Comment (laurit) [inline instrumentation/dropwizard/dropwizard-views-0.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/dropwizardviews/ViewRenderTest.java:9]:
```suggestion
package io.opentelemetry.javaagent.instrumentation.dropwizardviews;
```

File: instrumentation/dropwizard/dropwizard-views-0.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/dropwizardviews/ViewRenderTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    View view = new View("/views/ftl/utf8.ftl", StandardCharsets.UTF_8) {};
+    new FreemarkerViewRenderer().render(view, Locale.ENGLISH, outputStream);
+    Thread.sleep(500);
+    assertEquals(0, testing.spans().size());
+  }
+
+  @AfterEach
+  void cleanUp() {
+    testing.clearData();
```
Comment (laurit) [inline instrumentation/dropwizard/dropwizard-views-0.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/dropwizardviews/ViewRenderTest.java:74]: why is this needed?

File: instrumentation/dropwizard/dropwizard-views-0.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/dropwizardviews/ViewRenderTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  @MethodSource("provideParameters")
+  void testSpan(ViewRenderer renderer, String template) throws IOException {
+    View view = new View(template, StandardCharsets.UTF_8) {};
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    testing.runWithSpan(
+        "parent",
+        () -> {
+          renderer.render(view, Locale.ENGLISH, outputStream);
+        });
+    assertTrue(outputStream.toString("UTF-8").contains("This is an example of a view"));
```
Comment (laurit) [inline instrumentation/dropwizard/dropwizard-views-0.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/dropwizardviews/ViewRenderTest.java:53]: You could use assertj `assertThat(outputStream.toString("UTF-8")).contains("This is an example of a view")`

File: instrumentation/dropwizard/dropwizard-views-0.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/dropwizardviews/ViewRenderTest.java (hunk 2 above)
Comment (quinoant) [reply instrumentation/dropwizard/dropwizard-views-0.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/dropwizardviews/ViewRenderTest.java:74]: This allows each of the iterations of the test to start with a clean slate rather than a trace with other spans that had been created by previous test interfering. Taking that part out causes every test after the first one to fail.

Comment (laurit) [reply instrumentation/dropwizard/dropwizard-views-0.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/dropwizardviews/ViewRenderTest.java:74]: Perhaps this is so because you have not added `@RegisterExtension` annotation on `InstrumentationExtension` field?

File: instrumentation/dropwizard/dropwizard-views-0.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/dropwizardviews/ViewRenderTest.java (hunk 3 above)
Comment (quinoant) [reply instrumentation/dropwizard/dropwizard-views-0.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/dropwizardviews/ViewRenderTest.java:51]: Great recommendation.  I will apply that suggestion.

**Comment by quinoant** [comment]
Thanks for the tips and the quick review!

**Review by mateuszrzeszutek — APPROVED** [review]
LGTM 👍 
Thanks @quinoant !
